woocommerce/woocommerce-android
# Issue/5356 Advanced error handling

Closes: #5356


### Description

The PR implements error handling similar to [the iOS one](https://github.com/woocommerce/woocommerce-ios/pull/5511). It uses shorter titles though.

It also contains a small refactoring of the `CardReaderPaymentViewModel` 

_Sorry for the big PR. Mostly it testing code though_

### Testing instructions

* Try to collect test amounts (201, 101, 102, 103 etc cents).
* Notice error messages
* Try to pay for an order < 50 cents
* Notice error message

### Images/gif

![image](https://user-images.githubusercontent.com/4923871/144043168-6df6a75c-8326-4c63-aade-9f20160dde10.png)
![image](https://user-images.githubusercontent.com/4923871/144043238-8f96f68f-450d-4407-b2b5-4d8e687aafd6.png)
![image](https://user-images.githubusercontent.com/4923871/144044051-ac1b3c9d-6631-4525-a434-289b8e41de14.png)
